src/oas3/OpenApi.ts

```diff
@@ -95,7 +96,7 @@ export default class OpenApi implements ApiInterface<OAS3ApiInfo> {
 
                 if (operation && contentType) {
                     mediaType = operation.getRequestMediaType(contentType);
-                    if (!mediaType) {
+                    if (!mediaType && httpHasBody(headers)) {
```

Comment:
What if this is a post with an invalid contentType and no body?  I feel like we shouldn't just pass that through.  I feel like there should be a check to see if this is a GET or a DELETE somewhere.  :P

Reply:
Actually, we should definitely not let a POST, PUT, etc... with an invalid content-type through.  Suppose you had a POST endpoint which didn't really need a body (like say a /api/:postId/markAsSeen).  If you require an empty body of content type "application/json" (e.g. `{}`), this can be used as a defense against CSRF attacks, since forms will never generate a content type of application/json.  But, if you created a form like:

```
```

Reply:
Oh, I see. Ok, I'll change it so it ignores the content-type for the following methods: GET, HEAD, TRACE, OPTIONS. [According to MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods) DELETE may have a request body.